### PR TITLE
Rem 265

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
fi-tax-schedules

### CODE REVIEW

File: src/pages/fi-tax-schedules/forms/DetailsForm.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            amount: yup.string().required(' '),
+            taxBase: yup.string().required(' '),
+
+            taxCodeId: yup.string().required(' ')
+          })
+        )
+        .required(' ')
+    }),
+
+    initialValues: {
```
Comment (RebeccaFrem02) [inline src/pages/fi-tax-schedules/forms/DetailsForm.js:39]: can we move initial values up?

File: src/pages/fi-tax-schedules/forms/DetailsForm.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            }))
+          } else {
+            formik.setValues({ TaxDetail: [{ id: 1, taxId: '', taxBase: '', taxCodeId: '', amount: '' }] })
+          }
+        })
+        .catch(error => {})
+  }, [recordId])
+
+  return (
+    <>
```
Comment (RebeccaFrem02) [inline src/pages/fi-tax-schedules/forms/DetailsForm.js:115]: is this <> necessary?

File: src/pages/fi-tax-schedules/forms/DetailsForm.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                },
+
+                {
+                  component: 'numberfield',
+                  label: labels.amount,
+                  name: 'amount',
+                  decimalScale: 2
+                }
+              ]}
+              height={`${expanded ? height - 280 : 380}px`}
```
Comment (RebeccaFrem02) [inline src/pages/fi-tax-schedules/forms/DetailsForm.js:171]: why are we still sending height here?

File: src/pages/fi-tax-schedules/Window/TaxSchedulesWindow.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+// ** Custom Imports
```
Comment (RebeccaFrem02) [inline src/pages/fi-tax-schedules/Window/TaxSchedulesWindow.js:1]: this comment serve no purpose

File: src/pages/fi-tax-schedules/forms/DetailsForm.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+import { useWindowDimensions } from 'src/lib/useWindowDimensions'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { DataSets } from 'src/resources/DataSets'
+
+const DetailsForm = ({ store, setStore, maxAccess, labels, expanded, editMode }) => {
+  const { recordId } = store
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { height } = useWindowDimensions()
```
Comment (hadichahine) [inline src/pages/fi-tax-schedules/forms/DetailsForm.js:18]: why?

File: src/pages/fi-tax-schedules/forms/DetailsForm.js [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+import { ResourceIds } from 'src/resources/ResourceIds'
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+import { useWindowDimensions } from 'src/lib/useWindowDimensions'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { DataSets } from 'src/resources/DataSets'
+
+const DetailsForm = ({ store, setStore, maxAccess, labels, expanded, editMode }) => {
```
Comment (hadichahine) [inline src/pages/fi-tax-schedules/forms/DetailsForm.js:15]: remove unused params

File: src/pages/fi-tax-schedules/forms/DetailsForm.js [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    recordId &&
+      getRequest({
+        extension: FinancialRepository.TaxDetailPack.qry,
+        parameters: `_taxId=${recordId}`
+      })
+        .then(res => {
+          if (res?.list?.length > 0) {
+            const items = res.list.map((item, index) => ({
+              ...item,
+              id: index + 1,
```
Comment (hadichahine) [inline src/pages/fi-tax-schedules/forms/DetailsForm.js:95]: Why do we need tot do that?

File: src/pages/fi-tax-schedules/forms/DetailsForm.js [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+          TaxDetail: items
+        }))
+      })
+      .catch(error => {})
+  }
+
+  useEffect(() => {
+    const defaultParams = `_taxId=${recordId}`
+    var parameters = defaultParams
+    recordId &&
```
Comment (hadichahine) [inline src/pages/fi-tax-schedules/forms/DetailsForm.js:86]: if(record)

File: src/pages/fi-tax-schedules/forms/DetailsForm.js [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+              amount: item.amount,
+              taxCodeName: item.taxCodeName
+            }))
+            formik.setValues({ TaxDetail: items })
+            setStore(prevStore => ({
+              ...prevStore,
+              TaxDetail: items
+            }))
+          } else {
+            formik.setValues({ TaxDetail: [{ id: 1, taxId: '', taxBase: '', taxCodeId: '', amount: '' }] })
```
Comment (hadichahine) [inline src/pages/fi-tax-schedules/forms/DetailsForm.js:107]: We already have initialvalues I guess.

File: src/pages/fi-tax-schedules/forms/TaxSchedulesForm.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { MasterSource } from 'src/resources/MasterSource'
+
+export default function TaxSchedulesForm({ labels, maxAccess, setStore, store }) {
+  const { recordId } = store
+  const [editMode, setEditMode] = useState(!!recordId)
```
Comment (hadichahine) [inline src/pages/fi-tax-schedules/forms/TaxSchedulesForm.js:18]: Remove if not needed

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/fi-tax-schedules/Window/TaxSchedulesWindow.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
+import CustomTabPanel from 'src/components/Shared/CustomTabPanel'
+import { CustomTabs } from 'src/components/Shared/CustomTabs'
+import { useState } from 'react'
+import TaxSchedulesForm from '../forms/TaxSchedulesForm'
+
+import DetailsForm from '../forms/DetailsForm'
+
+const TaxSchedulesWindow = ({ height, recordId, labels, maxAccess, expanded }) => {
```
[the remark this answers is not in the document]
Comment (Alinadimmokdad) [reply src/pages/fi-tax-schedules/Window/TaxSchedulesWindow.js:8]: no

File: src/pages/fi-tax-schedules/forms/DetailsForm.js (hunk 9 above)
Comment (Alinadimmokdad) [reply src/pages/fi-tax-schedules/forms/DetailsForm.js:107]: yes i am set the Values

File: src/pages/fi-tax-schedules/forms/TaxSchedulesForm.js [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+        formik.setValues({
+          ...obj,
+          recordId: response.recordId
+        })
+      } else toast.success('Record Edited Successfully')
+
+      invalidate()
+    }
+  })
+  const editMode = !!formik.values.recordId || !!recordId
```
Comment (hadichahine) [inline src/pages/fi-tax-schedules/forms/TaxSchedulesForm.js:57]: We're already setting the store recordId. we can stick to !!recordId. I guess editMode is passed from the parent too.

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/fi-tax-schedules/forms/DetailsForm.js (hunk 5 above)
Comment (hadichahine) [reply src/pages/fi-tax-schedules/forms/DetailsForm.js]: remove

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Review by hadichahine — APPROVED** [review]

**Pull request merged**